src/sqlfluff/core/parser/segments/base.py

```diff
@@ -10,13 +10,20 @@
 
 from io import StringIO
 from benchit import BenchIt
-from typing import Optional
+from cached_property import cached_property

```

Comment:
I tried to test the `sqlfluff fix` performance issue (#368) on this PR, but I got a `ModuleNotFoundError` error here. Maybe you forgot to commit a new file, `cached_property.py`?

This is the detailed error message:
```
time sqlfluff fix --rules L001,L003,L006,L008,L010,L011,L016,L018,L019,L022,L023,L030 slow_query.sql
Traceback (most recent call last):
  File "/Users/bhart/.pyenv/versions/sqlfluff-3.7.3/bin/sqlfluff", line 11, in <module>
    load_entry_point('sqlfluff', 'console_scripts', 'sqlfluff')()
  File "/Users/bhart/.pyenv/versions/3.7.3/envs/sqlfluff-3.7.3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/Users/bhart/.pyenv/versions/3.7.3/envs/sqlfluff-3.7.3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2793, in load_entry_point
    return ep.load()
  File "/Users/bhart/.pyenv/versions/3.7.3/envs/sqlfluff-3.7.3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2411, in load
    return self.resolve()
  File "/Users/bhart/.pyenv/versions/3.7.3/envs/sqlfluff-3.7.3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2417, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/__init__.py", line 5, in <module>
    from .api import lint, fix, parse  # noqa: F401
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/api/__init__.py", line 6, in <module>
    from .simple import lint, fix, parse
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/api/simple.py", line 3, in <module>
    from ..core import Linter
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/core/__init__.py", line 9, in <module>
    from .linter import Linter
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/core/linter.py", line 19, in <module>
    from .parser import Lexer, Parser
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/core/parser/__init__.py", line 5, in <module>
    from .segments import (
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/core/parser/segments/__init__.py", line 5, in <module>
    from .base import BaseSegment, UnparsableSegment
  File "/Users/bhart/dev/sqlfluff/src/sqlfluff/core/parser/segments/base.py", line 13, in <module>
    from cached_property import cached_property
ModuleNotFoundError: No module named 'cached_property'
```

Reply:
Did you re-install the `requirements.txt` from this branch? `cached_property` is a new dependency

Reply:
@alanmcruickshank Should `cached-property` be added to `setup.py` as a new `sqlfluff` dependency?

Reply:
Good catch.

Reply:
@barrywhart - I think this problem is resolved now. Can you try again?

Ideally, I'm most interested in `parse` and `lint` commands for this PR. Speed improvements for `fix` are coming with the source mapping that I'm working on right now.

Reply:
I’ll try again on Monday. Previously, I reran `pip install -e .`. Note that the entire slow query is in the ticket, so y'all could time it, too.

Reply:
I ran `sqlfluff lint` on the large query in #368. It's about 14% faster. Since there's a lot of output, I tested it both writing to the screen and writing to `/dev/null`. Performance was very similar both ways.

Reply:
Awesome. Thanks for the review 👍  . I'll compare the query myself on the next ticket.